src/main/cpp/pop/PopulationBuilder.cpp

```diff
@@ -85,32 +92,51 @@ shared_ptr<Population> PopulationBuilder::Build(
-                        StringUtils::FromString<unsigned int>(values[0]),
-                        StringUtils::FromString<unsigned int>(values[1]),
-                        StringUtils::FromString<unsigned int>(values[2]),
-                        StringUtils::FromString<unsigned int>(values[3]),
-                        StringUtils::FromString<unsigned int>(values[4]),
-                        StringUtils::FromString<unsigned int>(values[5]),
-                        start_infectiousness, start_symptomatic, time_infectious, time_symptomatic));
-                ++person_id;
+        if (boost::algorithm::ends_with(file_name, ".csv")) {
+                // Read population data file.
+                string line;
+                getline(pop_file, line); // step over file header
+                unsigned int person_id = 0U;
+                while (getline(pop_file, line)) {
+                        // Make use of stochastic disease characteristics.
+                        const auto start_infectiousness = Sample(rng, distrib_start_infectiousness);
+                        const auto start_symptomatic    = Sample(rng, distrib_start_symptomatic);
+                        const auto time_infectious      = Sample(rng, distrib_time_infectious);
+                        const auto time_symptomatic     = Sample(rng, distrib_time_symptomatic);
+                        const auto values = StringUtils::Split(line, ",");
+                        population.emplace_back(Person(
+                                person_id,
+                                StringUtils::FromString<unsigned int>(values[0]), // age
+                                StringUtils::FromString<unsigned int>(values[1]), // household_id
+                                StringUtils::FromString<unsigned int>(values[2]), // school_id
+                                StringUtils::FromString<unsigned int>(values[3]), // work_id
+                                StringUtils::FromString<unsigned int>(values[4]), // primary_community_id
+                                StringUtils::FromString<unsigned int>(values[5]), // secondary_community_id
+                                start_infectiousness,
+                                start_symptomatic,
+                                time_infectious,
+                                time_symptomatic));
+                        ++person_id;
+                }
+        } else if (boost::algorithm::ends_with(file_name, ".xml")) {
+                // Read population model file.
+                using boost::property_tree::ptree;
+                ptree pt;
+                read_xml(pop_file, pt);
+                PopulationModel model{ pt };
+
+                // Example:
+                std::cout << "child_maximum_age_gap: " << model.child_maximum_age_gap << std::endl;
+                // TODO: generate `population` from `model`.
+
+        } else {
+                throw runtime_error(string(__func__)
```

Comment:
Are we okay with relying solely on file extensions for file type detection? Perhaps a switch to override that could be useful (sort of like `gcc`/`clang`'s `-x c++ <filename>` syntax). Do you think that'd be a lot of work?

Anyway, I don't think this is of much importance. Just keep the code as-is if you think adding an override switch will take more than five minutes. This isn't worth a lot of effort.